src/web_app/js/background.js

```diff
@@ -9,14 +9,149 @@
-    'width': 800,
-    'height': 600,
-    'left': 0,
-    'top': 0
+(function() {
+  chrome.app.runtime.onLaunched.addListener(function() {
+    chrome.app.window.create('appwindow.html', {
+      'width': 800,
+      'height': 600,
+      'left': 0,
+      'top': 0
+    });
   });
-});
+
+  // Send message from background to window.
+  var sendMessageToWindow = function(port, message) {
+    trace('B -> W: ' + JSON.stringify(message));
+    try {
+      port.postMessage(message);
+    }
+    catch (ex) {
+      trace('Error sending message: ' + ex);
+    }
+  };
+
+  var handleWebSocketRequest = function(port, message) {
+    if (message.wsAction === Constants.WS_CREATE_ACTION) {
+      trace('RWS: creating web socket: ' + message.wssUrl);
+      var ws = new WebSocket(message.wssUrl);
+      ws.onopen = function() {
+        trace('RWS: onopen');
+        sendMessageToWindow(port, {
+          action: Constants.WS_ACTION,
+          wsAction: Constants.EVENT_ACTION,
+          wsEvent: Constants.WS_EVENT_ONOPEN
+        });
+      };
+      ws.onerror = function() {
+        trace('RWS: onerror.');
+        sendMessageToWindow(port, {
+          action: Constants.WS_ACTION,
+          wsAction: Constants.EVENT_ACTION,
+          wsEvent: Constants.WS_EVENT_ONERROR
+        });
+      };
+      ws.onclose = function(event) {
+        trace('RWS: onclose - code: ' + event.code + ' reason: ' +
+            event.reason);
+        sendMessageToWindow(port, {
+          action: Constants.WS_ACTION,
+          wsAction: Constants.EVENT_ACTION,
+          wsEvent: Constants.WS_EVENT_ONCLOSE,
+          data: event
+        });
+      };
+      ws.onmessage = function(event) {
+        trace('RWS: onmessage: ' + event.data);
+        sendMessageToWindow(port, {
+          action: Constants.WS_ACTION,
+          wsAction: Constants.EVENT_ACTION,
+          wsEvent: Constants.WS_EVENT_ONMESSAGE,
+          data: event
+        });
+      };
+      port.webSocket_ = ws;
+    } else if (message.wsAction === Constants.WS_SEND_ACTION) {
+      trace('RWS: sending: ' + message.data);
+      port.webSocket_.send(message.data);
+    } else if (message.wsAction === Constants.WS_CLOSE_ACTION) {
+      trace('RWS: close');
+      port.webSocket_.close();
+      port.webSocket = null;
+    }
+  };
+
+  var createQueuePromise = function(port, message) {
+    return function() {
+      trace('Executing queue action: ' + JSON.stringify(message));
+      if (message.action === Constants.XHR_ACTION) {
+        var method = message.method;
+        var url = message.url;
+        var body = message.body;
+        return sendAsyncUrlRequest(method, url, body);
+      } else if (message.action === Constants.WS_ACTION) {
+        handleWebSocketRequest(port, message);
+      } else {
+        trace('Unknown action in cleanup queue: ' + message.action);
+      }
+    };
+  };
+
+  var createQueueCatch = function() {
+    return function(error) {
```

Comment:
One of our jslint/jscs tools doesn't allow creating methods in a loop, to avoid modified closures. It isn't smart enough to detect that you aren't using any variables from the closure. This is the simplest way to make the rule happy. The other option is to disable the rule for that block.